packages/flutter_tools/test/src/process.dart

```diff
@@ -0,0 +1,144 @@
+/// for the sake of a clean appearance, when doing so meets the requirements.
+/// For example, `shellEscapeString("asdf") == "asdf"`.
+///
+/// See also [shellEscapeCommand], for operating on a whole command line.
+String shellEscapeString(String value) {
+  if (_definitelyShellLiteralWordRegExp.hasMatch(value)) {
+    return value;
+  }
+  return "'${value.replaceAll("'", r"'\''")}'";
+}
+
+/// A string which a shell would parse as the given command.
+///
+/// Useful for printing a command unambiguously, or for printing
+/// a command the user might want to copy-paste and run.
+///
+/// This method makes some effort to print the command's elements
+/// verbatim, for the sake of a clean appearance, where possible.
+/// For example, `shellEscapeCommand(['git', 'commit', '-am', 'a commit'])`
+/// returns `git commit -am 'a commit'`.
+///
+/// See also [shellEscapeString], for operating on an individual
+/// argument of a command.
+String shellEscapeCommand(List<String> command) {
+  return command.map(shellEscapeString).join(' ');
+}
+
+extension ProcessManagerExtension on ProcessManager {
+  /// Start a process and run it to completion, throwing an exception on failure.
+  ///
+  /// Like [runSync], this blocks until the child process terminates.
+  ///
+  /// If the child process exits with failure (a nonzero [ProcessResult.exitCode]),
+  /// this method throws an exception with details of the command and its output.
+  ProcessResult runSyncSuccess(
+    List<String> command, {
+    String? workingDirectory,
+    Map<String, String>? environment,
+    bool includeParentEnvironment = true,
+    // no runInShell; keep that always false
+    Encoding? stdoutEncoding = systemEncoding,
+    Encoding? stderrEncoding = systemEncoding,
+  }) {
+    final ProcessResult result = runSync(
+      command,
+      workingDirectory: workingDirectory,
+      environment: environment,
+      includeParentEnvironment: includeParentEnvironment,
+      stdoutEncoding: stdoutEncoding,
+      stderrEncoding: stderrEncoding,
+    );
+    if (result.exitCode != 0) {
+      throw Exception(
+        'child process exited with code ${result.exitCode}\n'
+        'command: ${shellEscapeCommand(command)}\n'
+        'stdout: ================================================================\n'
+        '${result.stdout}\n'
+        'stderr: ================================================================\n'
+        '${result.stderr}\n'
+        'end ====================================================================',
+      );
+    }
+    return result;
+  }
+}
+
+final RegExp _trailingNewlineRegExp = RegExp(r'\n*$');
+
+String _asShellOutput(String raw) {
+  return raw.replaceFirst(_trailingNewlineRegExp, '');
+}
+
+List<int> _bytesAsShellOutput(List<int> raw) {
+  final int end = 1 + raw.lastIndexWhere((int byte) => byte != 0x0a);
+  return end == raw.length ? raw : raw.sublist(0, end);
+}
+
+extension FileExtension on File {
```

Comment:
Can you make these non-extension helper methods? https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#avoid-using-extension